sdk/devcenter/developer-devcenter-rest/src/azureDevCenter.ts

```diff
@@ -19,28 +20,5 @@ export default function createClient(
   devCenterDnsSuffix: string = "devcenter.azure.com",
   options: ClientOptions = {}
 ): AzureDevCenterClient {
-  const baseUrl = options.baseUrl ?? `https://${tenantId}-${devCenter}.${devCenterDnsSuffix}`;
```

Comment:
I think this is a good time to add an overload for createClient that takes no `devCenterDnsSuffix` think about the scenario where a customer wants to use de default value for `devCenterDnsSuffix` and also pass some options. They will be forced to either type the default value or pass an `undefined` 
```ts
createClient("TenantId", "devCenter", creds, undefined, {apiVersion: "v1.1"})
```

If you define this with an overload


```ts
function createClient(
  tenantId: string,
  devCenter: string,
  credentials: TokenCredential,
  options?: ClientOptions
): AzureDevCenterClient;
function createClient(
  tenantId: string,
  devCenter: string,
  credentials: TokenCredential,
  devCenterDnsSuffix: string,
  options?: ClientOptions
): AzureDevCenterClient;
function createClient(
  tenantId: string,
  devCenter: string,
  credentials: TokenCredential,
  devCenterDnsSuffixOrOptions?: string | ClientOptions,
  options: ClientOptions = {}
): AzureDevCenterClient
```

Reply:
@joheredi took a stab at this. It's a bit different from your initial suggestion here, so let me know what you think!

Reply:
@joheredi one other consideration - since this modified developer-devcenter.api.md, does this necessitate another API review?

Reply:
I had another idea I wanted to run by folks, for RLC can we still augment ClientOptions? e.g.

```ts
export interface DevCenterClientOptions extends ClientOptions {
   devCenterDnsSuffix?: string;
}

export default function createClient(
  tenantId: string,
  devCenter: string,
  credentials: TokenCredential,
  options?: ClientOptions
): AzureDevCenterClient;
```

That way we can keep all the options in one options bag and not worry about overloads. I'm thinking of the case where a dev is trying to wrap our client and maybe sometimes has a dns suffix and maybe not (say they retrieve that from AppConfig or something) without having to do an if/else to explicitly call one function or the other.

@sebrenna yes changing the `api.md` file will trigger another review, but you should be able to merge this and it will automagically show up in APIView and one of us can approve it without needing to go to the board or anything that heavy for such a small change.

Reply:
@xirzec cool! I requested review from everyone in APIView if you didn't already have the link.

RE: placing `devCenterDnsSuffix` in options object - @chrissmiller and I would prefer to keep `devCenterDnsSuffix` as its own argument in this version for the following reasons:

* API with `devCenterDnsSuffix` has already been approved for prior checkins.
* Want to minimize churn where possible before release.
* In the next beta version of this package, we're going to get rid of `tenantId` + `devCenter` + `devCenterDnsSuffix` entirely by converging to the more typical Azure SDK behavior of accepting an `endpoint` argument for the resource. Therefore, we'd prefer to invest in customizations related to this code as little as necessary.